packages/core/src/routes/applications/application.ts

```diff
@@ -261,6 +261,20 @@ export default function applicationRoutes<T extends AuthedRouter>(
         }
       }
 
+      if (protectedAppMetadata) {
+        const { type, protectedAppMetadata: originProtectedAppMetadata } =
+          await findApplicationById(id);
+        assertThat(type === ApplicationType.Protected, 'application.protected_application_only');
+        assertThat(originProtectedAppMetadata, 'application.protected_application_misconfigured');
+        await updateApplicationById(id, {
+          protectedAppMetadata: {
+            ...originProtectedAppMetadata,
+            ...protectedAppMetadata,
+          },
+        });
+        await protectedApps.syncAppConfigsToRemote(id);
```

Comment:
A similar question, if sync AppConfigs failed, should we revert the updates?

Reply:
updated